simtools/model/mirrors.py

```diff
@@ -54,26 +56,25 @@ def _read_mirror_list_from_ecsv(self):
         InvalidMirrorListFile
             If number of mirrors is 0.
         """
+        # Getting mirror parameters from mirror list.
 
-        # TODO - temporary hard wired geopars - should come from DB
-        self.diameter = 120
-        self.shape = 1
-        self._logger.debug(f"Shape = {self.shape}")
-        self._logger.debug(f"Diameter = {self.diameter}")
-
-        _mirror_table = Table.read(self._mirror_list_file, format="ascii.ecsv")
+        self.mirror_table = Table.read(self._mirror_list_file, format="ascii.ecsv")
         self._logger.debug(f"Reading mirror properties from {self._mirror_list_file}")
         try:
-            self._mirrors["flen"] = list(_mirror_table["mirror_panel_radius"].to("cm").value / 2.0)
-            self.number_of_mirrors = len(self._mirrors["flen"])
-            self._mirrors["number"] = list(range(self.number_of_mirrors))
-            self._mirrors["pos_x"] = [0.0] * self.number_of_mirrors
-            self._mirrors["pos_y"] = [0.0] * self.number_of_mirrors
-            self._mirrors["diameter"] = [self.diameter] * self.number_of_mirrors
-            self._mirrors["shape"] = [self.shape] * self.number_of_mirrors
+            self.shape_type = u.Quantity(self.mirror_table["shape_type"])[0].value
+            self.mirror_diameter = u.Quantity(self.mirror_table["mirror_diameter"])[0].value
+            self.number_of_mirrors = len(self.mirror_table["focal_length"])
+
+            self._logger.debug(f"Mirror shape_type = {self.shape_type}")
+            self._logger.debug(f"Mirror diameter = {self.mirror_diameter}")
+            self._logger.debug(f"Number of Mirrors = {self.number_of_mirrors}")
         except KeyError:
-            self._logger.debug(
-                f"Missing column for mirror panel focal length (flen) in {self._mirror_list_file}"
+            self._logger.debug("Mirror shape_type not in mirror file")
+            self.number_of_mirrors = len(self.mirror_table["mirror_id"])
+
+        if "focal_length" not in self.mirror_table.colnames:
+            self.mirror_table["focal_length"] = (
```

Comment:
Wait, that's not correct. The focal length (for spherical mirrors) is half the **radius of curvature**, not the mirror radius. For example, the diameter of each LST mirror is 151 cm, but their focal length is roughly 2900 cm.
Instead, you should set the focal length of each mirror to zero. This will tell sim_telarray take the nominal focal length and randomise around it with `RANDOM_FOCAL_LENGTH`